browser/ui/views/location_bar/brave_news_location_view.cc

```diff
@@ -10,210 +10,154 @@
-#include "ui/views/layout/flex_layout_view.h"
-#include "ui/views/layout/layout_types.h"
-#include "ui/views/view.h"
+#include "ui/views/bubble/bubble_dialog_delegate_view.h"
 
 namespace {
+
 constexpr SkColor kSubscribedLightColor = SkColorSetRGB(76, 84, 210);
 constexpr SkColor kSubscribedDarkColor = SkColorSetRGB(115, 122, 222);
 
-class BraveNewsButtonView : public views::LabelButton,
-                            public TabStripModelObserver,
-                            public BraveNewsTabHelper::PageFeedsObserver {
- public:
-  BraveNewsButtonView(Profile* profile, TabStripModel* tab_strip)
-      : views::LabelButton(
-            base::BindRepeating(&BraveNewsButtonView::ButtonPressed,
-                                base::Unretained(this))),
-        profile_(profile),
-        tab_strip_(tab_strip) {
-    DCHECK(profile_);
-    SetAccessibleName(
-        l10n_util::GetStringUTF16(IDS_BRAVE_NEWS_ACTION_VIEW_TOOLTIP));
-    SetHorizontalAlignment(gfx::HorizontalAlignment::ALIGN_CENTER);
-
-    auto* ink_drop = views::InkDrop::Get(this);
-    ink_drop->SetMode(views::InkDropHost::InkDropMode::ON);
-    ink_drop->SetBaseColorCallback(base::BindRepeating(
-        [](views::View* host) { return GetToolbarInkDropBaseColor(host); },
-        this));
-    ink_drop->SetVisibleOpacity(kToolbarInkDropVisibleOpacity);
-    SetHasInkDropActionOnClick(true);
-
-    tab_strip_->AddObserver(this);
-    if (tab_strip_->GetActiveWebContents()) {
-      BraveNewsTabHelper::FromWebContents(tab_strip_->GetActiveWebContents())
-          ->AddObserver(this);
-    }
+}  // namespace
 
-    should_show_.Init(brave_news::prefs::kShouldShowToolbarButton,
-                      profile->GetPrefs(),
-                      base::BindRepeating(&BraveNewsButtonView::Update,
-                                          base::Unretained(this)));
-    opted_in_.Init(brave_news::prefs::kBraveTodayOptedIn, profile->GetPrefs(),
-                   base::BindRepeating(&BraveNewsButtonView::Update,
-                                       base::Unretained(this)));
-    news_enabled_.Init(brave_news::prefs::kNewTabPageShowToday,
-                       profile->GetPrefs(),
-                       base::BindRepeating(&BraveNewsButtonView::Update,
-                                           base::Unretained(this)));
-
-    auto menu_button_controller = std::make_unique<views::MenuButtonController>(
-        this,
-        base::BindRepeating(&BraveNewsButtonView::ButtonPressed,
-                            base::Unretained(this)),
-        std::make_unique<views::Button::DefaultButtonControllerDelegate>(this));
-    SetButtonController(std::move(menu_button_controller));
-
-    Update();
+BraveNewsLocationView::BraveNewsLocationView(
+    Profile* profile,
+    IconLabelBubbleView::Delegate* icon_label_bubble_delegate,
+    PageActionIconView::Delegate* page_action_icon_delegate)
+    : PageActionIconView(nullptr,
+                         0,
```

Comment:
I think tagging this args with param name would be good for readers.
```suggestion
    : PageActionIconView(/*command_updater=*/ nullptr,
                         /*command_id=*/0,
```

Reply:
Fair. Fixed.

Reply:
That's fair. Fixed.